java/compression/src/test/java/org/apache/arrow/compression/TestCompressionCodecFile.java

```diff
@@ -0,0 +1,176 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.compression;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.BitVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.compression.CompressionCodec;
+import org.apache.arrow.vector.compression.CompressionUtil;
+import org.apache.arrow.vector.compression.NoCompressionCodec;
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.ArrowFileWriter;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.ipc.ArrowStreamReader;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestCompressionCodecFile {
+  private final CompressionCodec codec;
+  private final int vectorLength;
+  private final File file;
+
+  public TestCompressionCodecFile(CompressionUtil.CodecType type, int vectorLength, CompressionCodec codec,
+                                  File file) {
+    this.codec = codec;
+    this.file = file;
+    this.vectorLength = vectorLength;
+  }
+
+  @Parameterized.Parameters(name = "codec = {0}, length = {1}, file = {2}")
+  public static Collection<Object[]> getCodecs() {
+    List<Object[]> params = new ArrayList<>();
+
+    int[] lengths = new int[]{10, 100, 1000};
+    for (int len : lengths) {
+      CompressionCodec dumbCodec = NoCompressionCodec.INSTANCE;
+      File fileNoCompression = new File("target/write_no_compression_" + len + ".arrow");
+      params.add(new Object[]{dumbCodec.getCodecType(), len, dumbCodec, fileNoCompression});
+
+      CompressionCodec lz4Codec = new Lz4CompressionCodec();
+      File fileLZ4Compression = new File("target/write_lz4_compression_" + len + ".arrow");
```

Comment:
```suggestion
      File fileLz4Compression = new File("target/write_lz4_compression_" + len + ".arrow");
```
nit: generally we use camel case for abbreviates as well